<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>446</width>
    <height>689</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>446</width>
    <height>689</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>446</width>
    <height>689</height>
   </size>
  </property>
  <property name="palette">
   <palette>
    <active>
     <colorrole role="Button">
      <brush brushstyle="LinearGradientPattern">
       <gradient startx="0.000000000000000" starty="1.000000000000000" endx="0.000000000000000" endy="0.000000000000000" type="LinearGradient" spread="PadSpread" coordinatemode="ObjectBoundingMode">
        <gradientstop position="0.000000000000000">
         <color alpha="255">
          <red>10</red>
          <green>10</green>
          <blue>10</blue>
         </color>
        </gradientstop>
        <gradientstop position="1.000000000000000">
         <color alpha="255">
          <red>20</red>
          <green>20</green>
          <blue>25</blue>
         </color>
        </gradientstop>
       </gradient>
      </brush>
     </colorrole>
     <colorrole role="Base">
      <brush brushstyle="LinearGradientPattern">
       <gradient startx="0.000000000000000" starty="1.000000000000000" endx="0.000000000000000" endy="0.000000000000000" type="LinearGradient" spread="PadSpread" coordinatemode="ObjectBoundingMode">
        <gradientstop position="0.000000000000000">
         <color alpha="255">
          <red>10</red>
          <green>10</green>
          <blue>10</blue>
         </color>
        </gradientstop>
        <gradientstop position="1.000000000000000">
         <color alpha="255">
          <red>20</red>
          <green>20</green>
          <blue>25</blue>
         </color>
        </gradientstop>
       </gradient>
      </brush>
     </colorrole>
     <colorrole role="Window">
      <brush brushstyle="LinearGradientPattern">
       <gradient startx="0.000000000000000" starty="1.000000000000000" endx="0.000000000000000" endy="0.000000000000000" type="LinearGradient" spread="PadSpread" coordinatemode="ObjectBoundingMode">
        <gradientstop position="0.000000000000000">
         <color alpha="255">
          <red>10</red>
          <green>10</green>
          <blue>10</blue>
         </color>
        </gradientstop>
        <gradientstop position="1.000000000000000">
         <color alpha="255">
          <red>20</red>
          <green>20</green>
          <blue>25</blue>
         </color>
        </gradientstop>
       </gradient>
      </brush>
     </colorrole>
    </active>
    <inactive>
     <colorrole role="Button">
      <brush brushstyle="LinearGradientPattern">
       <gradient startx="0.000000000000000" starty="1.000000000000000" endx="0.000000000000000" endy="0.000000000000000" type="LinearGradient" spread="PadSpread" coordinatemode="ObjectBoundingMode">
        <gradientstop position="0.000000000000000">
         <color alpha="255">
          <red>10</red>
          <green>10</green>
          <blue>10</blue>
         </color>
        </gradientstop>
        <gradientstop position="1.000000000000000">
         <color alpha="255">
          <red>20</red>
          <green>20</green>
          <blue>25</blue>
         </color>
        </gradientstop>
       </gradient>
      </brush>
     </colorrole>
     <colorrole role="Base">
      <brush brushstyle="LinearGradientPattern">
       <gradient startx="0.000000000000000" starty="1.000000000000000" endx="0.000000000000000" endy="0.000000000000000" type="LinearGradient" spread="PadSpread" coordinatemode="ObjectBoundingMode">
        <gradientstop position="0.000000000000000">
         <color alpha="255">
          <red>10</red>
          <green>10</green>
          <blue>10</blue>
         </color>
        </gradientstop>
        <gradientstop position="1.000000000000000">
         <color alpha="255">
          <red>20</red>
          <green>20</green>
          <blue>25</blue>
         </color>
        </gradientstop>
       </gradient>
      </brush>
     </colorrole>
     <colorrole role="Window">
      <brush brushstyle="LinearGradientPattern">
       <gradient startx="0.000000000000000" starty="1.000000000000000" endx="0.000000000000000" endy="0.000000000000000" type="LinearGradient" spread="PadSpread" coordinatemode="ObjectBoundingMode">
        <gradientstop position="0.000000000000000">
         <color alpha="255">
          <red>10</red>
          <green>10</green>
          <blue>10</blue>
         </color>
        </gradientstop>
        <gradientstop position="1.000000000000000">
         <color alpha="255">
          <red>20</red>
          <green>20</green>
          <blue>25</blue>
         </color>
        </gradientstop>
       </gradient>
      </brush>
     </colorrole>
    </inactive>
    <disabled>
     <colorrole role="Button">
      <brush brushstyle="LinearGradientPattern">
       <gradient startx="0.000000000000000" starty="1.000000000000000" endx="0.000000000000000" endy="0.000000000000000" type="LinearGradient" spread="PadSpread" coordinatemode="ObjectBoundingMode">
        <gradientstop position="0.000000000000000">
         <color alpha="255">
          <red>10</red>
          <green>10</green>
          <blue>10</blue>
         </color>
        </gradientstop>
        <gradientstop position="1.000000000000000">
         <color alpha="255">
          <red>20</red>
          <green>20</green>
          <blue>25</blue>
         </color>
        </gradientstop>
       </gradient>
      </brush>
     </colorrole>
     <colorrole role="Base">
      <brush brushstyle="LinearGradientPattern">
       <gradient startx="0.000000000000000" starty="1.000000000000000" endx="0.000000000000000" endy="0.000000000000000" type="LinearGradient" spread="PadSpread" coordinatemode="ObjectBoundingMode">
        <gradientstop position="0.000000000000000">
         <color alpha="255">
          <red>10</red>
          <green>10</green>
          <blue>10</blue>
         </color>
        </gradientstop>
        <gradientstop position="1.000000000000000">
         <color alpha="255">
          <red>20</red>
          <green>20</green>
          <blue>25</blue>
         </color>
        </gradientstop>
       </gradient>
      </brush>
     </colorrole>
     <colorrole role="Window">
      <brush brushstyle="LinearGradientPattern">
       <gradient startx="0.000000000000000" starty="1.000000000000000" endx="0.000000000000000" endy="0.000000000000000" type="LinearGradient" spread="PadSpread" coordinatemode="ObjectBoundingMode">
        <gradientstop position="0.000000000000000">
         <color alpha="255">
          <red>10</red>
          <green>10</green>
          <blue>10</blue>
         </color>
        </gradientstop>
        <gradientstop position="1.000000000000000">
         <color alpha="255">
          <red>20</red>
          <green>20</green>
          <blue>25</blue>
         </color>
        </gradientstop>
       </gradient>
      </brush>
     </colorrole>
    </disabled>
   </palette>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <property name="styleSheet">
   <string notr="true">QWidget{
background: solid qlineargradient(spread:pad, x1:0, y1:1, x2:0, y2:0, stop:0 rgba(10, 10, 10, 255), stop:1 rgba(20, 20, 25, 255));
border: none;
border-radius: 15px;
}

QScrollArea{
border: 1px solid transparent;
border-radius: 20px;
}
QPushButton {
background-color: rgba(255,255,255,10);
color: #ffffff;
border: 1px solid transparent;
border-radius: 12px;
padding: 10px 20px 10px 20px;
}
QPushButton:hover {
background-color: rgba(255,255,255,20);
}
QPushButton:pressed {
background-color: rgba(255,255,255,10);
}
QPushButton:disabled {
background-color: #222222;
color: #666666;
}
QPushButton:focus {
outline: none;
border: 1px solid #0066FF;
}
QListView {
background-color: rgba(255,255,255,10);
border: 1px solid transparent;
border-radius: 15px;
padding: 5px;
}
QListView::item {
border: none;
padding: 5px;
margin: 5px;
color: #ffffff;
}
QListView::item:selected {
background-color: #0066FF;
border: 1px solid transparent;
border-radius: 5px;
color: #ffffff;
}
QListView::item:selected:focus {
border: 1px solid #0066FF;
}
QListView::item:hover {
background-color: rgba(0,102,255,20);
border: 1px solid transparent;
border-radius: 5px;
color: #0066FF;
}
QListView::item:disabled {
color: #666666;
background-color: #222222;
}
QLineEdit {
background-color: rgba(255,255,255,10);
border: 1px solid transparent;
border-radius: 10px;
padding: 8px;
color: #ffffff;
}
QLineEdit:focus {
border: 1px solid #0066FF;
}
QLineEdit::placeholder {
color: #666666;
}
QLineEdit:disabled {
background-color: #222222;
color: #666666;
border: 1px solid transparent;
}
QProgressBar {
border: 1px solid transparent;
height: 12px;
border-radius: 6px;
background-color: rgba(255,255,255,10);
text-align: center;
}
QProgressBar::chunk {
background-color: #00DC72;
border: 1px solid transparent;
border-radius: 6px;
}
QProgressBar::indicator {
border: 1px solid transparent;
border-radius: 5px;
}
QProgressBar:disabled {
border: 1px solid transparent;
background-color: #222222;
}
QProgressBar:focus {
border: 1px solid transparent;
}
QCheckBox {
background-color: transparent;
color: #ffffff;
border: 1px solid transparent;
border-radius: 5px;
}
QCheckBox::indicator {
width: 18px;
height: 18px;
margin-left: 5px;
}
QRadioButton {
color: #ffffff;
border: none;
}
QRadioButton::indicator {
width: 15px;
height: 15px;
margin-left: 5px;
}
QTabWidget::pane {
border: 1px solid transparent;
border-radius: 20px;
background-color: rgba(255,255,255,10);
}
QTabBar::tab {
color: white;
border: 1px solid transparent;
border-bottom-color: rgba(255,255,255,10);
border-top-left-radius: 15px;
min-width: 80px;
border-top-right-radius: 15px;
padding: 6px;
margin-right: 2px;
}
QTabBar::tab:selected {
background-color: rgba(255,255,255,10);
border-bottom: 2px solid #00DC72;
color: white;
}
QTabBar::tab:!selected {
background-color: rgba(255,255,255,20);
color: #dddddd;
}
QTabBar::tab:hover {
background-color: rgba(255,255,255,20);
}
QTabBar::tab:pressed {
background-color: rgba(255,255,255,10);
}
QTabBar::tab:disabled {
background-color: rgba(255,255,255,30);
color: #666666;
}
QComboBox {
background-color: rgba(255,255,255,10);
border: 1px solid transparent;
border-radius: 10px;
padding: 8px;
color: #ffffff;
}
QComboBox:hover {
background-color: rgba(255,255,255,20);
}
QComboBox:disabled {
background-color: rgba(255,255,255,30);
color: #666666;
}
QComboBox:pressed {
background-color: rgba(255,255,255,10);
}
QComboBox QAbstractItemView {
background: solid transparent;
border: 1px solid transparent;
border-radius: 5px;
}
QComboBox::item {
padding:5px 0px 5px 0px;
max-height: 18px;
}
QComboBox::item:selected {
background: solid #0066FF;
border: 1px solid transparent;
border-radius: 5px;
}
QGroupBox {
background-color: rgba(255,255,255,10);
border: 1px solid transparent;
border-radius: 15px;
padding: 10px;
margin: 20px;
}
QGroupBox::title {
margin-top: -20px;
subcontrol-position: top;
padding: 3px 15px 5px 15px;
color: #ffffff;
background-color: #222222;
border: 1px solid transparent;
border-radius: 10px;
}

QTableWidget {
background-color: rgba(255,255,255,10);
gridline-color: rgba(255,255,255,30);
border: 1px solid transparent;
border-radius: 15px;
}

QTableWidget QHeaderView{
background-color:  transparent;
}

QHeaderView::section::horizontal {
background-color:  transparent;
color: #ffffff;
padding: 4px;
}

QHeaderView::section::vertical {
background-color:  transparent;
color: #ffffff;
padding: 4px;
}

QTableCornerButton::section {
background-color: transparent;
}

QTableWidget::item {
background-color: rgba(255,255,255,10);
padding: 7px;
border: 1px solid transparent;
}

QTableWidget::item:alternate {
background-color: rgba(255,255,255,30);
}

QTableWidget::item:selected {
border-bottom: 2px solid #00DC72;
color: white;
}

QTableWidget::item:hover {
background-color: rgba(255,255,255,20);
color: white;
}

QScrollBar:vertical {
background-color: transparent;
width: 10px;
}

QScrollBar::handle:vertical {
background-color: rgba(255,255,255,30);
border: 1px solid transparent;
border-radius: 5px;
min-height: 20px;
}

QScrollBar::add-line:vertical, QScrollBar::sub-line:vertical {
background-color: transparent;
}

QScrollBar:horizontal {
background-color: transparent;
width: 10px;
}

QScrollBar::handle:horizontal {
background-color: rgba(255,255,255,30);
border: 1px solid transparent;
border-radius: 5px;
min-height: 20px;
}

QScrollBar::add-line:horizontal, QScrollBar::sub-line:horizontal {
background-color: transparent;
}

QLabel{
background-color: transparent;
}</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>40</x>
      <y>40</y>
      <width>361</width>
      <height>561</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Vazir</family>
      <pointsize>14</pointsize>
      <bold>false</bold>
     </font>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout">
     <property name="spacing">
      <number>15</number>
     </property>
     <item>
      <widget class="QLabel" name="label_2">
       <property name="font">
        <font>
         <family>Vazir</family>
         <pointsize>24</pointsize>
         <bold>false</bold>
        </font>
       </property>
       <property name="text">
        <string>Custom DNS for MacOS</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignmentFlag::AlignCenter</set>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="verticalSpacer">
       <property name="font">
        <font>
         <family>Vazir</family>
         <bold>false</bold>
        </font>
       </property>
       <property name="orientation">
        <enum>Qt::Orientation::Vertical</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>20</width>
         <height>40</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QLabel" name="label">
       <property name="font">
        <font>
         <family>Vazir</family>
         <pointsize>14</pointsize>
         <bold>false</bold>
        </font>
       </property>
       <property name="text">
        <string>Enter DNS</string>
       </property>
      </widget>
     </item>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout">
       <item>
        <widget class="QLineEdit" name="edtDns">
         <property name="font">
          <font>
           <family>Vazir</family>
           <pointsize>14</pointsize>
           <bold>false</bold>
          </font>
         </property>
         <property name="styleSheet">
          <string notr="true"/>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="btnAddList">
         <property name="font">
          <font>
           <family>Vazir</family>
           <pointsize>14</pointsize>
           <bold>false</bold>
          </font>
         </property>
         <property name="styleSheet">
          <string notr="true"/>
         </property>
         <property name="text">
          <string>Add</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <widget class="QListWidget" name="lstDns">
       <property name="font">
        <font>
         <family>Vazir</family>
         <pointsize>14</pointsize>
         <bold>false</bold>
        </font>
       </property>
       <property name="styleSheet">
        <string notr="true"/>
       </property>
       <property name="editTriggers">
        <set>QAbstractItemView::EditTrigger::AnyKeyPressed|QAbstractItemView::EditTrigger::DoubleClicked|QAbstractItemView::EditTrigger::EditKeyPressed</set>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="label_3">
       <property name="font">
        <font>
         <family>Vazir</family>
         <pointsize>14</pointsize>
         <bold>false</bold>
        </font>
       </property>
       <property name="text">
        <string>Select DNS</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QComboBox" name="cbDns">
       <property name="font">
        <font>
         <family>Vazir</family>
         <pointsize>14</pointsize>
         <bold>false</bold>
        </font>
       </property>
       <property name="styleSheet">
        <string notr="true"/>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="verticalSpacer_2">
       <property name="font">
        <font>
         <family>Vazir</family>
         <pointsize>14</pointsize>
         <bold>false</bold>
        </font>
       </property>
       <property name="orientation">
        <enum>Qt::Orientation::Vertical</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>20</width>
         <height>40</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QPushButton" name="btnAddDns">
       <property name="font">
        <font>
         <family>Vazir</family>
         <pointsize>14</pointsize>
         <bold>false</bold>
        </font>
       </property>
       <property name="styleSheet">
        <string notr="true"/>
       </property>
       <property name="text">
        <string>Set DNS</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="btnClearDns">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Minimum" vsizetype="Expanding">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="font">
        <font>
         <family>Vazir</family>
         <pointsize>14</pointsize>
         <bold>false</bold>
        </font>
       </property>
       <property name="styleSheet">
        <string notr="true"/>
       </property>
       <property name="text">
        <string>Clear DNS</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QStatusBar" name="statusbar">
   <property name="styleSheet">
    <string notr="true"/>
   </property>
  </widget>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>edtDns</sender>
   <signal>returnPressed()</signal>
   <receiver>btnAddList</receiver>
   <slot>click()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>187</x>
     <y>202</y>
    </hint>
    <hint type="destinationlabel">
     <x>367</x>
     <y>205</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
